ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/ScanConfigActionsColumn.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { ReactElement } from 'react';
+import { generatePath, useHistory } from 'react-router-dom';
+import { ActionsColumn } from '@patternfly/react-table';
+
+import useFeatureFlags from 'hooks/useFeatureFlags';
+import { ComplianceScanConfigurationStatus } from 'services/ComplianceScanConfigurationService';
+
+import { scanConfigDetailsPath } from './compliance.scanConfigs.routes';
+
+// Component for scan configs table table page corresponds to ScanConfigActionDropdown for scan config details page.
+
+// Caller is responsible for conditional rendering only if READ_WRITE_ACCESS level for Compliance resource.
+
+export type ScanConfigActionsColumnProps = {
+    handleDeleteScanConfig: (scanConfigResponse: ComplianceScanConfigurationStatus) => void;
+    handleRunScanConfig: (scanConfigResponse: ComplianceScanConfigurationStatus) => void;
+    handleSendReport: (scanConfigResponse: ComplianceScanConfigurationStatus) => void;
+    scanConfigResponse: ComplianceScanConfigurationStatus;
+};
+
+function ScanConfigActionsColumn({
+    handleDeleteScanConfig,
+    handleRunScanConfig,
+    handleSendReport,
+    scanConfigResponse,
+}: ScanConfigActionsColumnProps): ReactElement {
+    const history = useHistory();
+    const { isFeatureFlagEnabled } = useFeatureFlags();
+    const isComplianceReportingEnabled = isFeatureFlagEnabled('ROX_COMPLIANCE_REPORTING');
+
+    const { id, lastExecutedTime, scanConfig } = scanConfigResponse;
+    const { notifiers } = scanConfig;
+    const scanConfigUrl = generatePath(scanConfigDetailsPath, {
+        scanConfigId: id,
+    });
+    const isScanning = lastExecutedTime === null;
+
+    const items = [
+        {
+            title: 'Edit scan schedule',
+            description: isScanning ? 'Edit is disabled while scan is running' : '',
+            isDisabled: isScanning,
+            onClick: (event) => {
+                event.preventDefault();
+                history.push({
+                    pathname: scanConfigUrl,
+                    search: 'action=edit',
+                });
+            },
+        },
+        {
+            isSeparator: true,
+        },
+        {
+            title: 'Run scan now',
+            description: isScanning ? 'Run is disabled while scan is already running' : '',
+            isDisabled: isScanning,
+            onClick: (event) => {
+                event.preventDefault();
+                handleRunScanConfig(scanConfigResponse);
+            },
+        },
+        /* eslint-disable no-nested-ternary */
```

Comment:
Yes, a good item to discuss at next team meeting and plan any change in 4.6A sprint.

Because of many years work with descriptive linguists who write rules as if nested ternary, I have had to stretch beyond my preference to empathize with many who dislike them so strongly.